core/src/main/resources/error/error-classes.json

```diff
@@ -122,6 +166,9 @@
     "message" : [ "Invalid pivot value '%s': value data type %s does not match pivot column data type %s" ],
     "sqlState" : "42000"
   },
+  "READING_AMBIGUOUS_DATES" : {
```

Comment:
For these upgrade class names, should we always include the Spark version name? Maybe `READING_AMBIGUOUS_NAMES_IN_SPARK3`